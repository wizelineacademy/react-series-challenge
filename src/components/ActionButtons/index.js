import AddFavButton from './addFavButton';
import RemoveFavButton from './removeFavButton';

export {
    AddFavButton,
    RemoveFavButton
}
