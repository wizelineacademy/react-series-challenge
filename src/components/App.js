import React from "react";
import { Route, Switch, Redirect } from "react-router-dom";
import HomePage from "./pages/HomePage/index";
import FavsPage from "./pages/FavsPage/index";

const App = () => {
  return (
    <Switch>
      <Route exact path="/" render={() => <Redirect to="/home" />} />
      <Route path="/home" component={HomePage} />;
      <Route path="/favorites" component={FavsPage} />;
    </Switch>
  );
};

export default App;
