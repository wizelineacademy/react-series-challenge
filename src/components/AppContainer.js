import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Route, Switch } from 'react-router-dom';
import { bindActionCreators } from 'redux';

import gifActions from '../actions/gifActions';

import GifGrid from '../components/GifGrid';
import SearchBar from '../components/SearchBar';

import { ContainerWrapper } from '../styles/ContainerWrapper.style.js';


export class AppContainer extends Component {
  componentDidMount() {
    const { getTrendingGifs } = this.props;

    getTrendingGifs();
  }

  render() {
    const { gifs, favorites } = this.props;

    return (
      <ContainerWrapper className="mt">
        <SearchBar />

        <Switch>
          <Route
            exact
            path="/"
            render={({ match }) => (
              <GifGrid gifs={ gifs } />
            )}
          />
          <Route
            path="/favorites"
            render={({ match }) => (
              <GifGrid gifs={ favorites } />
            )}
          />
        </Switch>
      </ContainerWrapper>
    );
  }
}

const mapStateToProps = (state) => state;

const mapDispatchToProps = (dispatch) => {
  const { getTrendingGifs } = gifActions.creators;

  return bindActionCreators({ getTrendingGifs }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(AppContainer);
