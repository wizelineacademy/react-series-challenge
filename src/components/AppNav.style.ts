import styled from 'styled-components';
import { NavLink as RouterNavLink } from 'react-router-dom';

export const StyledNav = styled.nav`
    border-bottom: 1px solid #FC4349;
    margin-bottom: 1em;
    color: #6DBCDB;
    font-weight: bold;

    ul {
        display: flex;
        justify-content: space-around;
        text-align: center;
        align-items: center;
        height: 3.5em;
    }

    li { flex: 1; }

    a {
        text-decoration: none;
        color: inherit;
        padding: 1em;
    }
`;

export const NavLink = styled(RouterNavLink)`
    &.active { color: #FC4349; }
`
