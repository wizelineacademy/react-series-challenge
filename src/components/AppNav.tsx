import * as React from 'react';
import ROUTES from '../routes';
import Search from './Search';

import { StyledNav, NavLink } from './AppNav.style'

const LINKS = ROUTES.map(({path, content}) =>
    <li key={path}><NavLink to={path}>{content}</NavLink></li>);

class AppNav extends React.Component {
    public render() {
        return (<StyledNav>
            <ul>
                {LINKS}
                <li><Search /></li>
            </ul>
        </StyledNav>);
    }
}

export default AppNav;
