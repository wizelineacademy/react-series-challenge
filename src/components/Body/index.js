import React from 'react';
import './styles.css';

const Body = ({ children }) => {
  return (
    <div className="Body">
      { children }
    </div>
  );
};

export default Body;
