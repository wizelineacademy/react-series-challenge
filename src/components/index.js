import Card from './Card'
import Detail from './Detail'
import GifItem from './GifItem'
import Grid from './Grid'
import Header from './Header'
import Home from './Home'
import SearchBar from './SearchBar'

export {
  Card,
  Detail,
  GifItem,
  Grid,
  Header,
  Home,
  SearchBar
}
