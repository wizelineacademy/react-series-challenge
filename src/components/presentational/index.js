import Menu from './Menu';
import ItemMenu from './ItemMenu';
import GiphyList from './GiphyList';
import InputSearch from './InputSearch';
import ErrorWatcher from './ErrorWatcher';

export {
  Menu,
  ItemMenu,
  GiphyList,
  InputSearch,
  ErrorWatcher
};