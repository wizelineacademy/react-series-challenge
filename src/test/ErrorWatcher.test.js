import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import {ErrorWatcher} from '../components/presentational';

describe('ErrorWatcher', () => {
  it('should render correctly', () => {
    const output = shallow(
      <ErrorWatcher />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('Set visible true', () => {
    const wrapper = shallow(
      <ErrorWatcher visible={true} />
    );
    const section = wrapper.find('section');
    expect(section.exists()).toEqual(true);
  });
});
