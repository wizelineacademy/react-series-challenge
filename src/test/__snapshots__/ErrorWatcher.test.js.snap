// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorWatcher should render correctly 1`] = `
<section
  className="sc-kAzzGY mksuQ"
/>
`;
